File: src/461.cpp
class Solution {
public:
    /*
    time defeat: 100%
    space defeat: 64.4%
    */
    int hammingDistance(int x, int y) {
        //然后用C++搞一遍
        //想通了，按位异或
        int bResult = x ^ y;
        //bitset长度设定为32的原因是x, y最大为2^31次方
        return bitset<32>(bResult).count();
    }
};

File: src/461.py
class Solution:
    '''
    time defeat: 30.34%
    space defeat: 28.37%
    '''
    def hammingDistance(self, x: int, y: int) -> int:
        #这个用python倒是很好做，直接转然后遍历就可以了
        bX = bin(x)[2:]
        bY = bin(y)[2:]
        #然后补全对齐
        #先保证bX永远是短的那个
        if len(bX) > len(bY):
            bX, bY = bY, bX
        while len(bX) < len(bY):
            bX = "0" + bX
        #然后按位比对
        result = 0
        for (i, j) in zip(bX, bY):
            if i != j:
                result += 1
            else:
                continue
        return result
